### PR TITLE
Set up VPC and RDS with Terraform

### LINKED ISSUE #20: Initialize Terraform project; add support for VPC and RDS
Initialize a Terraform project with remote state and instantiate:

- https://github.com/azavea/terraform-aws-vpc
- https://github.com/azavea/terraform-aws-postgresql-rds

### PR DESCRIPTION
**Pull request by coalkettler**
## Overview

Configures a base VPC and RDS instance. Introduces the `infra` script.

Resolves #20 

## Testing Instructions

* Run this command to get a shell in the Terraform container and use the staging Terraform config bucket:
  ```#bash
  NOAA_FLOOD_MAP_SETTINGS_BUCKET=noaafloodmap-staging-config-us-east-1 docker-compose -f docker-compose.ci.yml run --rm terraform
  ```
* Run `./scripts/plan` and inspect the Terraform plan output. The command should run successfully but no changes should be necessary.
* SSH into the staging bastion host. The PEM is available in an NOAA Flood Map subdirectory on the fileshare.
* Confirm that you can connect to the RDS instance with `psql`. The password is available in the `terraform.tfvars` file.

### CODE REVIEW

File: deployment/README.md
Comment (ColeKettler) [inline deployment/README.md:33]: Adding an extra layer of DNS namespacing in the event that we end up hosting more applications here at a later date.

File: deployment/terraform/database.tf
Comment (coalkettler) [inline deployment/terraform/database.tf]: All set to our common defaults that we use for new projects.

File: deployment/README.md
Comment (coalkettler) [inline deployment/README.md:35]: Adding an extra layer of DNS namespacing in the event that we end up hosting more applications here at a later date.

File: deployment/terraform/variables.tf
Comment (hectcastro) [inline deployment/terraform/variables.tf:153]: This variable exists at the project level, but it isn't being set at the module level.

File: deployment/terraform/variables.tf
Comment (hectcastro) [inline deployment/terraform/variables.tf]: I think that we should aim for PostgreSQL 12 here. This will likely necessitate updating the PostgreSQL client libraries on the bastion.

File: docker-compose.ci.yml
Comment (hectcastro) [inline docker-compose.ci.yml]: I'd be curious to use a more recent version of Terraform here. I created a PR to add the most recent 0.12.x patch release (0.12.28) to `docker-terraform` here: https://github.com/azavea/docker-terraform/pull/17.

**Review by hectcastro — COMMENTED** [review]
Everything looks good. A few minor comments regarding tooling and system version upgrades.

File: deployment/terraform/variables.tf
Comment (coalkettler) [reply deployment/terraform/variables.tf:153]: Shoot, thanks for catching that. Fixed!

File: docker-compose.ci.yml
Comment (coalkettler) [reply docker-compose.ci.yml]: Upgrading it to `0.12.29` after your most recent changes in that PR. I'll hold off on merging until the `docker-terraform` PR is merged and the image is available, I've got it built locally for now.

File: deployment/terraform/variables.tf
Comment (coalkettler) [reply deployment/terraform/variables.tf]: RDS instance has been updated and the Postgres 12 client libraries has been installed on the bastion. `amazon-linux-extras` doesn't have 12 yet so I installed via [Postgres' provided RPMs](https://download.postgresql.org/pub/repos/yum/12/redhat/rhel-7-x86_64/).

File: docker-compose.ci.yml
Comment (hectcastro) [reply docker-compose.ci.yml]: The PR has been merged and the 0.12.29 tag now exists on Quay.

A deprecated syntax warning pops up that's related to `terraform-aws-vpc`. I opened https://github.com/azavea/terraform-aws-vpc/issues/27 to address it, but it isn't fatal, so I don't think it blocks us here.

File: deployment/terraform/variables.tf
Comment (hectcastro) [reply deployment/terraform/variables.tf]: Were the RPMs installed in a way that allow the PostgreSQL packages to be upgraded via something like `yum update`? If not, I think the process you used to install it may be worth noting in this PR somewhere because that may become important if/when we need to upgrade in the future.

With RF, I ended up installing PostgreSQL client libraries from source because the PGDG RPMs didn't apply cleanly. Compiling from source isn't ideal, but knowing things were compiled from source helps make sense of the path to future upgrades.

**Review by hectcastro — APPROVED** [review]
👍

File: deployment/terraform/variables.tf
Comment (coalkettler) [reply deployment/terraform/variables.tf]: The `postgresql12` package can be managed by Yum. I don't think it will detect updates since the RPM was installed individually, rather than as part of a larger upstream repo.

If we need to upgrade in the future, I would hope that the `postgresql12` library would be available in `amazon-linux-extras` since RDS now supports 12. I've seen AWS support threads acknowledging this need but it seems to be in earlier stages of adoption.

I'll add a note to a README that details how it was installed and provides a link to where I retrieved the RPM.

**Pull request merged**
